src/Microsoft.OData.Core/Metadata/MetadataUtilsCommon.cs

```diff
@@ -125,6 +125,31 @@ internal static bool IsODataTypeDefinitionTypeKind(this IEdmType type)
             return type.TypeKind == EdmTypeKind.TypeDefinition;
         }
 
+        /// <summary>
+        /// Checks whether a type reference refers to an OData Untyped definition.
+        /// </summary>
+        /// <param name="typeReference">The (non-null) <see cref="IEdmTypeReference"/> to check.</param>
+        /// <returns>true if the <paramref name="typeReference"/> is an OData Untyped definition reference; otherwise false.</returns>
+        internal static bool IsODataUnTypedTypeKind(this IEdmTypeReference typeReference)
+        {
+            ExceptionUtils.CheckArgumentNotNull(typeReference, "typeReference");
+            ExceptionUtils.CheckArgumentNotNull(typeReference.Definition, "typeReference.Definition");
+
+            return typeReference.Definition.IsODataUnTypedTypeKind();
+        }
+
+        /// <summary>
+        /// Checks whether a type refers to an OData Untyped definition.
+        /// </summary>
+        /// <param name="type">The (non-null) <see cref="IEdmType"/> to check.</param>
+        /// <returns>true if the <paramref name="type"/> is an OData Untyped definition; otherwise false.</returns>
+        internal static bool IsODataUnTypedTypeKind(this IEdmType type)

```

Comment:
>IsODataUnTypedTypeKind [](start = 29, length = 22)

Do we have "IsUntyped(this IEdmType type)" extension method?

Reply:
I have added one as part of the LargeObjectStreaming PR.

---
In reply to: [266676418](https://github.com/OData/odata.net/pull/1322#discussion_r266676418) [](ancestors = 266676418)

Reply:
I tried looking for this in your LOS PR but I was unable to find it. 

---
In reply to: [267020044](https://github.com/OData/odata.net/pull/1322#discussion_r267020044) [](ancestors = 267020044,266676418)

Reply:
I tried finding this in your PR but I was unable to find it. Can you point me to the class to which this method was added?

---
In reply to: [267020044](https://github.com/OData/odata.net/pull/1322#discussion_r267020044) [](ancestors = 267020044,266676418)

Reply:
It's in EdmTypeSemantics.cs 

---
In reply to: [267131123](https://github.com/OData/odata.net/pull/1322#discussion_r267131123) [](ancestors = 267131123,267020044,266676418)